examples/traffic-light.manifold

```diff
@@ -0,0 +1,41 @@
+* = import Manifold.core
+* = import Manifold.digital
+
+Type TrafficLight = (
+  green:Boolean  = false,
+  yellow:Boolean = false,
+  red:Boolean    = false
+)
+
+GREEN_SECONDS  = 10
+YELLOW_SECONDS = 3
+TOTAL_SECONDS  = 2 * (GREEN_SECONDS + YELLOW_SECONDS)
+
+seconds = Manifold.Digital.count(hz: 1, mod: TOTAL_SECONDS)
+
+Manifold.Digital.if(
+  (seconds < GREEN_SECONDS, -> {
+    north_light = south_light = (green: true)
```

Comment:
I'll double check the spec in a second but my intention is to support `-> {}` notation wherever possible and unambiguous. 

These functions are in local function scope as they are passed as """closures""" to the `Manifold.Digital.if` component.